File: merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x:x[0])
        if len(intervals) > 0:
            ans = [intervals[0]]
        for i in range(1, len(intervals)):
            if (intervals[i][1] > ans[len(ans)-1][1]):
                if (ans[len(ans)-1][1] >= intervals[i][0]):
                    temp = ans.pop()
                    temp[1] = max(temp[1], intervals[i][1])
                    ans.append(temp)
                else:
                    ans.append(intervals[i])
        return ans

File: merge-intervals/merge-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    intervals.sort((a,b) => {
        if (a[0] === b[0]){
            return a[1]-b[1]
        }
        else {
            return a[0]-b[0]
        }
    })
    
    let i=1
    let res = [intervals[0]]
    while (i<intervals.length) {
        let last = res.pop()
        if (intervals[i][0] <= last[1]) {
            res.push([last[0], Math.max(intervals[i][1], last[1])])
        }
        else {
            res.push(last)
            res.push(intervals[i])
        }
        i++
    }
    return res
    
};